entproto/cmd/protoc-gen-entgrpc/validation.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019-present Facebook
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"entgo.io/contrib/entproto"
+	"entgo.io/ent/schema/field"
+	"google.golang.org/protobuf/compiler/protogen"
+)
+
+func typeNeedsValidator(d entproto.FieldMap) bool {
+	for _, fld := range d {
+		if fieldNeedsValidator(fld) {
+			return true
+		}
+	}
+	return false
+}
+
+func fieldNeedsValidator(d *entproto.FieldMappingDescriptor) bool {
+	if d.IsEdgeField {
+		return d.EdgeIDType() == field.TypeUUID
+	}
+	return d.EntField.IsUUID()
```

Comment:
Do you use the `EdgeIDType` method only here? Maybe you can do instead:

```suggestion
	f := d.EntField
	if d.IsEdgeField {
		f = d.EntEdge.Type.ID
	}
	return f.IsUUID()
```

Reply:
nice